cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/exercise/Sleep.scala

```diff
@@ -3,30 +3,143 @@ package com.ibm.sparktc.sparkbench.workload.exercise
 import com.ibm.sparktc.sparkbench.workload.{Workload, WorkloadDefaults}
 import com.ibm.sparktc.sparkbench.utils.GeneralFunctions._
 import org.apache.spark.sql.{DataFrame, SparkSession}
+import breeze.stats.distributions.{Poisson, Rand}
+import com.ibm.sparktc.sparkbench.utils.SparkBenchException
 
 case class SleepResult(
-                      name: String,
-                      timestamp: Long,
-                      total_runtime: Long
+                        name: String,
+                        timestamp: Long,
+                        total_runtime: Long
                       )
 
 object Sleep extends WorkloadDefaults {
+
+  def optionAnyToOptionLong(a: Option[Any]): Option[Long] = a match {
+    case None => None
+    case Some(any) => Some(any2Long(any))
+  }
+
+
   val name = "sleep"
-  def apply(m: Map[String, Any]) =
-    new Sleep(input = m.get("input").map(_.asInstanceOf[String]),
-      output = None,
-      sleepMS = (m.get("sleepms"), m.get("maxsleepms")) match {
-        case (Some(l), _) => any2Int2Long(l)
-        case (None, Some(l)) => randomLong(max = any2Int2Long(l))
-        case (_, _) => randomLong(max = 3600000L) //one hour
+  def apply(m: Map[String, Any]): Sleep = {
+
+    val sleepMS: Option[Long] = m.get("sleepms") match {
+      case None => None
+      case Some(any) => Some(any2Long(any))
+    }
+    val distribution: Option[String] = m.get("distribution").asInstanceOf[Option[String]]
+    val distributionMean: Option[Double] = m.get("mean").asInstanceOf[Option[Double]]
+    val distributionStd: Option[Double] = m.get("std").asInstanceOf[Option[Double]]
+    val distributionMin: Option[Long] = optionAnyToOptionLong(m.get("min"))
+    val distributionMax: Option[Long] = optionAnyToOptionLong(m.get("max"))
+
+    def buildWithDist(dist: String) = dist match {
+      case "uniform" => buildWithUniform(distributionMin, distributionMax)
+      case "poisson" => buildWithPoisson(distributionMean)
+      case "gaussian" => buildWithGaussian(distributionMean, distributionStd)
+      case other => throw SparkBenchException(
+        s"""
+           |Distribution $other is not implemented for the Sleep workload.
+           |Please see documentation for available distributions or specify "sleepMS".
+         """.stripMargin)
+    }
+
+    (sleepMS, distribution) match {
+      case (Some(sleep), Some(dist)) => throw  SparkBenchException("Cannot specify both sleepMS and a distribution. " +
+        "Please modify your config file.")
+      case (Some(sleep), None) => buildWithSleepMS(sleep)
+      case (None, None) => throw SparkBenchException("The Sleep workload requires either a time specified by sleepMS " +
+        "or a distribution from which to choose a time. Please modify your config file.")
+      case (None, Some(dist)) => buildWithDist(dist)
+    }
+
+  }
+
+  def buildWithSleepMS(long: Long) = long match {
```

Comment:
straight up if statement would be easier to read here imo

Reply:
Yeah, agree. Fixed.